File: leetcode/C++/Remove_duplicates_from_sorted_list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if(!head) return NULL;
        if(!(head->next)) return head;
        ListNode *cur = head;
        ListNode *curnext = head->next;
        
        while(curnext &&(cur->val) == (curnext->val)) 
            curnext = curnext->next;
        cur->next = deleteDuplicates(curnext);
        
        return cur;
    }
};

File: leetcode/linklist/ Remove Duplicates from Sorted List.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func deleteDuplicates(head *ListNode) *ListNode {
    c := head 
    
    for;c != nil;{
        nextNode := c.Next
        for;nextNode != nil && nextNode.Val == c.Val;{
            nextNode = nextNode.Next
        }
        c.Next = nextNode
        c = nextNode
    }

    return head
}